### PR TITLE
LPD-22571

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-22571

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-options-web/src/main/java/com/liferay/commerce/product/options/web/internal/display/context/CPSpecificationOptionDisplayContext.java [hunk 1]
```diff
@@ -107,16 +90,97 @@ public String getCPOptionCategoryTitle(
...
 		return StringPool.BLANK;
 	}
 
+	public CreationMenu getCreationMenu(
+		CPSpecificationOption cpSpecificationOption) {
+
+		return CreationMenuBuilder.addDropdownItem(
+			dropdownItem -> {
+				dropdownItem.setHref(
+					_getAddSpecificationOptionPicklistRenderURL(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-options-web/src/main/java/com/liferay/commerce/product/options/web/internal/display/context/CPSpecificationOptionDisplayContext.java:105]: `_getAddSpecificationOptionPicklistRenderURL` -> `_getAddSpecificationOptionListTypeDefinitionRenderURL`

File: modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/specification_option/add_existing_picklist.jsp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		context='<%=
+			HashMapBuilder.<String, Object>put(
+				"cmd", ParamUtil.getString(request, Constants.CMD)
+			).put(
+				"cpSpecificationOptionId", cpSpecificationOptionId
+			).put(
+				"namespace", liferayPortletResponse.getNamespace()
+			).build()
+		%>'
+		module="{CreateOrAssignCPSpecificationOptionPickList} from commerce-product-options-web"
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/specification_option/add_existing_picklist.jsp:36]: `CreateOrAssignCPSpecificationOptionPickList` -> `CreateOrAssignCPSpecificationOptionListTypeDefinition`

File: modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/specification_option/details.jsp [hunk 3]
```diff
@@ -63,10 +64,17 @@ List<CPOptionCategory> cpOptionCategories = cpSpecificationOptionDisplayContext.
...
+				HashMapBuilder.<String, Object>put(
+					"specificationId", cpSpecificationOption.getCPSpecificationOptionId()
+				).build()
+			%>'
+			apiURL='<%= "/o/headless-commerce-admin-catalog/v1.0/specifications/" + cpSpecificationOption.getCPSpecificationOptionId() + "/pick-lists" %>'
+			creationMenu="<%= cpSpecificationOptionDisplayContext.getCreationMenu(cpSpecificationOption) %>"
 			fdsActionDropdownItems="<%= cpSpecificationOptionDisplayContext.getFDSActionDropdownItems() %>"
 			id="<%= CommerceSpecificationOptionFDSNames.LIST_TYPE_DEFINITIONS %>"
 			itemsPerPage="<%= 10 %>"
+			propsTransformer="{CPSpecificationOptionPickListPropsTransformer} from commerce-product-options-web"
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/specification_option/details.jsp:77]: `CPSpecificationOptionPickListPropsTransformer` -> `CPSpecificationOptionListTypeDefinitionPropsTransformer`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 4]
```diff
@@ -901,6 +901,38 @@ components:
                 - key
                 - name
             type: object
+        PickList:
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:904]: `PickList` -> `ListTypeDefinition`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 5]
```diff
@@ -2141,6 +2173,11 @@ components:
                     type: string
                 optionCategory:
                     $ref: "#/components/schemas/OptionCategory"
+                pickListId:
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:2176]: `pickListId` -> `listTypeDefinitionId`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 6]
```diff
@@ -15412,6 +15449,158 @@ paths:
                       - HeadlessCommerceAdminCatalog.write
             summary: Updates a specification by ID.
             tags: ["Specification"]
+    "/specifications/{id}/pick-lists":
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:15452]: `pick-lists` -> `list-type-definitions`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 7]
```diff
@@ -15412,6 +15449,158 @@ paths:
...
+        get:
+            operationId: getSpecificationIdPickListsPage
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:15454]: Same here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 8]
```diff
@@ -15412,6 +15449,158 @@ paths:
...
+                        Unexpected error
+            security:
+                - BasicAuth: []
+                - BearerAuth: []
+                - OAuth2:
+                      - HeadlessCommerceAdminCatalog.read
+            summary: Gets the picklist associated to the specification.
+            tags: ["PickList"]
+        post:
+            operationId: postSpecificationIdPickList
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:15524]: Same here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 9]
```diff
@@ -15412,6 +15449,158 @@ paths:
...
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            security:
+                - BasicAuth: []
+                - BearerAuth: []
+                - OAuth2:
+                      - HeadlessCommerceAdminCatalog.write
+            summary: Creates a picklist and links it to the specification.
+            tags: ["PickList"]
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:15603]: `PickList` -> `ListTypeDefinition`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/PickListResourceImpl.java [hunk 10]
```diff
@@ -39,36 +39,47 @@ public Page<PickList> getSpecificationIdPickListsPage(Long id)
...
+			return super.getSpecificationIdPickListsPage(id);
+		}
+
+		ListTypeDefinition listTypeDefinition =
 			_listTypeDefinitionService.getListTypeDefinition(
-				cpSpecificationOption.getListTypeDefinitionId()))));
+				listTypeDefinitionId);
+
+		return Page.of(
+			Collections.singletonList(_toPickList(listTypeDefinition))
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/PickListResourceImpl.java]: `_toPickList` -> `_toListTypeDefinition`

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/ServiceProvider/commerce-admin-catalog/v1.0/SpecificationOption.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+const VERSION = 'v1.0';
+
+function resolvePath(basePath = '', id = '') {
+	return `${basePath}${VERSION}${ENTITY_PATH}/${id}`;
+}
+export default function SpecificationOption(basePath) {
+	return {
+		createSpecificationPicklistById: (specificationId, json) =>
+			AJAX.POST(
+				`${resolvePath(basePath, specificationId)}/pick-lists`,
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/ServiceProvider/commerce-admin-catalog/v1.0/SpecificationOption.js:19]: `pick-lists` -> `list-type-definitions`

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/ServiceProvider/commerce-admin-catalog/v1.0/SpecificationOption.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+				json
+			),
+		getSpecificationPicklistsById: (specificationId, json) =>
+			AJAX.GET(
+				`${resolvePath(basePath, specificationId)}/pick-lists`,
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/ServiceProvider/commerce-admin-catalog/v1.0/SpecificationOption.js:24]: `pick-lists` -> `list-type-definitions`

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Comment by andrea-ale-sbarra** [comment]
ci:report 19783169

**Pull request closed without merging**